src/TagInput/index.jsx

```diff
@@ -0,0 +1,363 @@
+    {
+        className        : undefined,
+        cssMap           : undefined,
+        dropdownPosition : 'bottom',
+        hasError         : false,
+        id               : undefined,
+        isDisabled       : false,
+        isReadOnly       : false,
+        onChange         : undefined,
+        placeholder      : undefined,
+        suggestions      : undefined,
+        tags             : undefined,
+    };
+
+    constructor( props )
+    {
+        super( props );
+
+        this.state = {
+            activeOption    : undefined,
+            filteredOptions : undefined,
+            id              : undefined,
+            inputValue      : '',
+            isOpen          : false,
+            tags            : undefined,
+        };
+
+        this.handleBlur            = this.handleBlur.bind( this );
+        this.handleChangeInput     = this.handleChangeInput.bind( this );
+        this.handleClickClose      = this.handleClickClose.bind( this );
+        this.handleClickOption     = this.handleClickOption.bind( this );
+        this.handleFocus           = this.handleFocus.bind( this );
+        this.handleKeyDown         = this.handleKeyDown.bind( this );
+        this.handleMouseOutOption  = this.handleMouseOutOption.bind( this );
+        this.handleMouseOverOption = this.handleMouseOverOption.bind( this );
+    }
+
+    static getDerivedStateFromProps( props, state )
+    {
+        const options =
+            normalizeOptions( props.suggestions ) || state.options || [];
+
+        return {
+            filteredOptions : state.inputValue ?
+                state.filteredOptions : options,
+            id   : props.id || state.id || generateId( 'TagInput' ),
+            options,
+            tags : props.tags || state.tags || [],
+        };
+    }
+
+    handleBlur()
+    {
+        this.setState( { isOpen: false } );
+    }
+
+    handleChangeInput( { value } )
+    {
+        this.setState( ( { options } ) =>
+        {
+            const filteredOptions = options.filter( ( { text } ) =>
+                text.match( new RegExp( value, 'i' ) ) );
+
+            const activeOption = ( value && filteredOptions.length ) ?
+                filteredOptions[ 0 ].id : undefined;
+
+            return {
+                activeOption,
+                filteredOptions,
+                inputValue : value,
+            };
+        } );
+    }
+
+    handleClickClose( { value } )
+    {
+        this.setState( ( { tags } ) => ( {
+            tags : tags.filter( tag => tag !== value ),
+        } ) );
+    }
+
+    handleClickOption( { id } )
+    {
+        this.setState( ( { filteredOptions, tags } ) =>
+        {
+            const option = getOption( id, filteredOptions );
+            const newTags = [ ...tags, option.text ];
+
+            const { onChange } = this.props;
+            if ( typeof onChange === 'function' )
+            {
+                onChange( { tags: newTags } );
+            }
```

Comment:
this should be also added to `handleClickClose` and `handleKeyDown`

Reply:
Thanks for catching!